src/ucp/core/ucp_ep.c

```diff
@@ -757,16 +756,28 @@ void ucp_ep_cleanup_lanes(ucp_ep_h ep)
     }
 }
 
+static int
+ucp_ep_local_disconnect_progress_remove_filter(const ucs_callbackq_elem_t *elem,
+                                               void *arg)
+{
+    ucp_ep_h ep = (ucp_ep_h)arg;
+    ucp_request_t *req;
+
+    if (elem->cb == ucp_ep_local_disconnect_progress) {
+        req = (ucp_request_t*)elem->arg;
+        if (ep == req->send.ep) {
```

Comment:
pls comment and assert (if possible) that this request was done to send EP_REMOVED message

Reply:
maybe we can check progress is wireup_msg_progress?

Reply:
> maybe we can check progress is wireup_msg_progress?

makes sense, done